### ISSUE TITLE
Spam "TypeError: no implicit conversion of nil into String"

### ISSUE THREAD

**Issue by Izzette** [newrelic/newrelic-ruby-agent#801]
## Description
Since 8.0.0, the NewRelic agent spams repeatedly many logs when NewRelic is not configured.  These two lines are repeated 10+ times per second.  None of these logs are generated and no reporting issue is observed when the agent is enabled.
```
** [NewRelic][2022-02-10 16:04:00 +0000 hostname (410)] WARN : Failed to create trace context payload
** [NewRelic][2022-02-10 16:04:00 +0000 hostname (410)] WARN : TypeError: no implicit conversion of nil into String
```

## Expected Behavior
Should not print error messages when agent is not configured.

## Logs
All newrelic logs up until the first pair of error messages:
```
** [NewRelic][2021-09-28 19:32:12 +0000 hostname (286)] WARN : No license key found. This often means your newrelic.yml file was not found, or it lacks a section for the running environment, 'production'. You may also want to try linting your newrelic.yml to ensure it is valid YML.
** [NewRelic][2021-09-28 19:32:18 +0000 hostname (309)] WARN : No license key found. This often means your newrelic.yml file was not found, or it lacks a section for the running environment, 'production'. You may also want to try linting your newrelic.yml to ensure it is valid YML.
** [NewRelic][2022-02-10 16:04:00 +0000 hostname (389)] WARN : Agent received a ForceDisconnectException from the server, disconnecting. (410: Gone)
** [NewRelic][2022-02-10 16:04:00 +0000 hostname (389)] WARN : Failed to create trace context payload
** [NewRelic][2022-02-10 16:04:00 +0000 hostname (389)] WARN : TypeError: no implicit conversion of nil into String
```
These logs are from our test environment (thus the jump to 2022-02-10) and crosses multiple process boundries.

## Steps to Reproduce
TODO: will start working on a minimal reproduction.

## Your Environment
Rails 6.1.4, ruby 3.0.2p107 (2021-07-07 revision 0db68f0233) [x86_64-linux]

**Comment by kaylareopelle** [comment]
Hi @Izzette! Thanks for bringing this to our attention. I tried to reproduce this with a Rails 6.1.4 app running on Ruby 3.0.2 that had version 8.0.0 of the agent installed, but no configuration file. Unfortunately, the same behavior did not come up. My logs stopped in a development environment at "Agent received a ForceDisconnect Exception..." and no logs were generated in the test environment.

When you have a chance to get the Steps to Reproduce filled out, we'll take another look!

**Comment by kaylareopelle** [comment]
Hi @Izzette  - Thanks for these additional details. I've reopened this issue and we'll take a look next week!

**Comment by kaylareopelle** [comment]
Hi @Izzette  - @tannalynn and I took a look at this issue today. We determined the root cause as the agent attempting to add distributed tracing headers before completing its connection to New Relic's servers, causing some required fields to be nil. Our solution is to stop distributed tracing headers from being added until the agent is connected. 

We're in the process of completing a release, so I don't believe this will be included as part of 8.2.0, but I will let you know if that changes. In the meantime, if you'd like to test out the fix, please update your Gemfile to point to this branch: 

```ruby 
gem 'newrelic_rpm', git: 'https://github.com/newrelic/newrelic-ruby-agent.git', branch: 'bugfix_trace_context_payload_typeerror'
```

We'd love to hear if this resolves your issue. 

Thanks for your support and patience!

**Comment by Izzette** [comment]
@kaylareopelle I have great news, it appears to work !

**Comment by kaylareopelle** [comment]
Hi @Izzette, I wanted to let you know this change has been included in our 8.2.0 release. You should be able to update your Gemfile to use this version. 

Thanks again for your help identifying and resolving this issue! 🌻
